src/brogue/Items.c

```diff
@@ -3536,59 +3545,121 @@ void negate(creature *monst) {
             }
         } else {
-            monst->status[STATUS_TELEPATHIC] = 0;
-            monst->status[STATUS_MAGICAL_FEAR] = 0;
-            monst->status[STATUS_LEVITATING] = 0;
+            if (monst->status[STATUS_TELEPATHIC] > 0 ) {
+                monst->status[STATUS_TELEPATHIC] = 0;
+                negated = true;
+            }
+            if (monst->status[STATUS_MAGICAL_FEAR] > 0 ) {
+                monst->status[STATUS_MAGICAL_FEAR] = 0;
+                negated = true;
+            }
+            if (monst->status[STATUS_LEVITATING] > 0 ) {
+                monst->status[STATUS_LEVITATING] = 0;
+                negated = true;
+            }
         }
-        monst->info.flags &= ~MONST_IMMUNE_TO_FIRE;
-        monst->movementSpeed = monst->info.movementSpeed;
-        monst->attackSpeed = monst->info.attackSpeed;
+        if (monst->info.flags & MONST_IMMUNE_TO_FIRE) {
+            monst->info.flags &= ~MONST_IMMUNE_TO_FIRE;
+            negated = true;
+        }
+        if (monst->movementSpeed != monst->info.movementSpeed) {
+            monst->movementSpeed = monst->info.movementSpeed;
+            negated = true;
+        }
+        if (monst->attackSpeed != monst->info.attackSpeed) {
+            monst->attackSpeed = monst->info.attackSpeed;
+            negated = true;
+        }
+
         if (monst != &player && monst->mutationIndex > -1 && mutationCatalog[monst->mutationIndex].canBeNegated
             && rogue.patchVersion >= 3) {
 
             monst->mutationIndex = -1;
+            negated = true;
         }
         if (monst != &player && (monst->info.flags & NEGATABLE_TRAITS)) {
             if ((monst->info.flags & MONST_FIERY) && monst->status[STATUS_BURNING]) {
                 extinguishFireOnCreature(monst);
             }
             monst->info.flags &= ~NEGATABLE_TRAITS;
+            negated = true;
             refreshDungeonCell(monst->xLoc, monst->yLoc);
             refreshSideBar(-1, -1, false);
         }
         for (i = 0; i < 20; i++) {
-            backupBolts[i] = monst->info.bolts[i];
-            monst->info.bolts[i] = BOLT_NONE;
-        }
-        for (i = 0, j = 0; i < 20 && backupBolts[i]; i++) {
-            if (boltCatalog[backupBolts[i]].flags & BF_NOT_NEGATABLE) {
-                monst->info.bolts[j] = backupBolts[i];
-                j++;
+            if (monst->info.bolts[i]) {
+                if (!(boltCatalog[monst->info.bolts[i]].flags & BF_NOT_NEGATABLE)) {
+                    monst->info.bolts[i] = BOLT_NONE;
+                    negated = true;
+                }
             }
         }
+
         monst->newPowerCount = monst->totalPowerCount; // Allies can re-learn lost ability slots.
         applyInstantTileEffectsToCreature(monst); // in case it should immediately die or fall into a chasm
     }
+
+    if (negated && monst != &player && !(monst->info.flags & MONST_DIES_IF_NEGATED)) {
+        sprintf(buf, "%s was negated", monstName);
```

Comment:
Brogue is a bit inconsistent but I think it generally prefers present tense. Also how about a more detailed description?

```suggestion
        sprintf(buf, "%s is stripped of its special traits", monstName);
```

Reply:
Sounds good to me!